gems/sorbet-runtime/lib/types/_types.rb

```diff
@@ -281,5 +281,8 @@ def self.[](type)
       end
     end
   end
+
+  # A special singleton used for static analysis of exceptions.
+  RETRY = Object.new
```

Comment:
Can you do something like this?

https://github.com/sorbet/sorbet/blob/234cbf9c047b524e7ac95ccaae706d0bee42ad49/gems/sorbet-runtime/lib/types/private/types/void.rb#L9-L18

There's context in the PR for why it's preferred:

#2585

it has to do with what reflection sees.

Reply:
Absolutely, thanks!